merlin/models/tf/models/ranking.py

```diff
@@ -204,32 +189,36 @@ def DeepFMModel(
     deep_block : Optional[Block]
         The `Block` that learns high-ordeer feature interactions
         Defaults to MLPBlock([64, 128])
+    input_block : Block, optional
+        The `Block` to use as the input layer. If None, a default `InputBlockV2` object
+        is instantiated, that creates the embedding tables for the categorical features
+        based on the schema, with the specified embedding_dim.
+        For a custom representation of input data you can instantiate
+        and provide an `InputBlockV2` instance.
     prediction_tasks: optional
         The prediction tasks to be used, by default this will be inferred from the Schema.
         Defaults to None
-    embedding_option_kwargs: Optional[dict]
-        Additional arguments to provide to `EmbeddingOptions` object
-        for embeddings tables setting.
-        Defaults to {}
     Returns
     -------
     Model
 
     """
-    input_block = InputBlock(
-        schema,
-        embedding_options=EmbeddingOptions(
-            embedding_dim_default=embedding_dim, **embedding_option_kwargs
-        ),
-        **kwargs,
-    )
+    input_block = input_block or InputBlockV2(schema, dim=embedding_dim, aggregation=None, **kwargs)
+
+    # TODO: Identify why we need to set the schema manually for `InputBlockV2`
+    # to avoid an error of ParallelBlock without schema?
+    input_block.set_schema(schema)
```

Comment:
@marcromeyn I had to include this hack because the `InputBlockV2` returns a `ParallelBlock` without schema in this case.  
I suspect that is the case because the data for this test contains only categorical features and the `Filter()` for continuous features gets empty, so that not all branches within `ParallelBlock` contains a schema.
Is this related to your recent refactory that makes `ParallelBlock` schema-aware?

Reply:
This hack is not needed anymore, as @marcromeyn implemented a fix for that on `InputBlockV2` in this [PR](https://github.com/NVIDIA-Merlin/models/pull/705)